osu.Game/Online/Chat/MessageFormatter.cs

```diff
@@ -116,70 +116,35 @@ public static LinkDetails GetLinkDetails(string url)
 
             switch (args[0])
             {
-                case "http":
-                case "https":
-                    // length > 3 since all these links need another argument to work
-                    if (args.Length > 3 && (args[1] == "osu.ppy.sh" || args[1] == "new.ppy.sh"))
+                case "http" when args.Length == 4 && (args[1] == "osu.ppy.sh" || args[1] == "new.ppy.sh"):
+                case "https" when args.Length == 4 && (args[1] == "osu.ppy.sh" || args[1] == "new.ppy.sh"):
```

Comment:
Opposed to this condition duplication.

Reply:
The duplication here eliminates the duplication of `External`. Duplicating here gives a weak positive favour to me.